cluster-autoscaler/config/kubernetes.go

```diff
@@ -113,3 +117,39 @@ func GetKubeClientConfig(uri *url.URL) (*kube_rest.Config, error) {
 
 	return kubeConfig, nil
 }
+
+// NewDynamicKubeConfigRoundTripper wraps the http.RoundTripper to provide the latest bearer token from the kube config on disk
+func NewDynamicKubeConfigRoundTripper(kubeconfigPath string, rt http.RoundTripper) http.RoundTripper {
+	return &bearerAuthRoundTripper{
+		kubeconfigPath: kubeconfigPath,
+		rt:             rt,
+	}
+}
+
+type bearerAuthRoundTripper struct {
+	kubeconfigPath string
+	rt             http.RoundTripper
+
+	lastModified time.Time
+	bearerToken  string
+}
+
+func (rt *bearerAuthRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	stat, err := os.Stat(rt.kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to stat kubeconfig path: %w", err)
+	}
+	if stat.ModTime().After(rt.lastModified) {
+		kubeConfig, err := kube_client_cmd.BuildConfigFromFlags("", rt.kubeconfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot build kube cluster config: %w", err)
+		}
+		rt.bearerToken = kubeConfig.BearerToken
```

Comment:
Loading the entire kubeconfig but only honoring the bearer token changing doesn't really make sense.  The bearer token should be in a distinct file that is reloaded, the rest of the contents should remain static.

Reply:
https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi#autoscaler-running-anywhere-with-separate-kubeconfigs-for-management-and-workload-clusters

Cluster Autoscaler can be configured in a mode where it is running in Cluster "1", but the cluster and resources it needs to scale and manage are in cluster "3", and the cloud configuration and node groups to scale are in cluster "2". In this scenario, there is no service account bearer token, except for running in Cluster "1" due to these being separate clusters.

The contents of the file we are loading is created by the CAPA awscluster controller, only has a bearer token, and is shaped like this:
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0t...
    server: https://XXXXX.us-east-2.eks.amazonaws.com
  name: cluster-name
contexts:
- context:
    cluster: cluster-3
    user: cluster-3-capi-admin
  name: cluster-3-capi-admin@cluster-3
current-context: cluster-3-capi-admin@cluster-3
kind: Config
preferences: {}
users:
- name: cluster-3-capi-admin
  user:
    token: k8s-aws-v1...
```

Are you suggesting that CAPA should encode a second secret value with just the `token` value contents above, and have the Cluster Autoscaler reload that contents, to simulate the behavior of a mounted service account bearer token? Or something else?

Reply:
Right if you change the YAML to:

```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: LS0t...
    server: https://XXXXX.us-east-2.eks.amazonaws.com
  name: cluster-name
contexts:
- context:
    cluster: cluster-3
    user: cluster-3-capi-admin
  name: cluster-3-capi-admin@cluster-3
current-context: cluster-3-capi-admin@cluster-3
kind: Config
preferences: {}
users:
- name: cluster-3-capi-admin
  user:
    tokenFile: location_of_file
```

And have the contents of `location_of_file` be:

```
k8s-aws-v1...
```

Then client-go will handle the contents of `location_of_file` changing for you.